### PR TITLE
Fix #channel on many structures not properly getting cached channels

### PR DESCRIPTION
**Pull request by TheEssem**
In some cases, for example when leaving/joining a voice channel, `VoiceState#channel` always returns `null` because `_cachedChannel` never gets defined. This should (hopefully) alleviate that.

### CODE REVIEW

**Comment by NotDemonix** [comment]
I had this issue too, Hope this will fix it

**Review by DonovanDMC — DISMISSED** [review]

File: lib/structures/VoiceState.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export default class VoiceState extends Base {
 
     /** The channel the user is connected to. */
     get channel(): VoiceChannel | StageChannel | null | undefined {
-        if (this.channelID !== null && this._cachedChannel !== null) {
+        if (this.channelID !== null && this._cachedChannel === null) {
```
Comment (coolcalcacol) [inline lib/structures/VoiceState.ts]: this will work for the first get then return null on any subsequent attempts afterwards.

**Review by coolcalcacol — CHANGES_REQUESTED** [review]

**Comment by coolcalcacol** [comment]
some other channel getters have the same issue, can you fix also?
GuildScheduledEvent, Webhook.

Invite's channel getter also looks a bit funky

File: lib/structures/VoiceState.ts (hunk 1 above)
Comment (DonovanDMC) [reply lib/structures/VoiceState.ts]: I planned on rewriting these getters to use `variable ??= value` instead of `variable ?? (varibable = value)`

Comment (Nuckyz) [reply lib/structures/VoiceState.ts]: You can actually just completely remove ` && this._cachedChannel === null`, it was exclusively for type safety and I didn't think of these issues when I made it

**Review by DonovanDMC — DISMISSED** [review]
LGTM, I'm going to wait for others to review before taking any action

File: lib/structures/Invite.ts [hunk 2]
```diff
@@ -144,14 +144,8 @@ export default class Invite<T extends InviteInfoTypes = "withMetadata", CH exten
...
     get channel(): (CH extends InviteChannel ? CH : PartialInviteChannel) | null {
-        if (this.channelID !== null && this._cachedChannel !== null) {
-            if (this._cachedChannel instanceof Channel) {
-                return this._cachedChannel;
-            }
-
-            const cachedChannel = this.client.getChannel<InviteChannel>(this.channelID);
-
-            return (cachedChannel ? (this._cachedChannel = cachedChannel as CH extends InviteChannel ? CH : PartialInviteChannel) : this._cachedChannel);
+        if (this.channelID !== null) {
```
Comment (Nuckyz) [inline lib/structures/Invite.ts:154]: You've completely removed the Partial/Full channel checking here

**Review by Nuckyz — CHANGES_REQUESTED** [review]

**Review by Nuckyz — APPROVED** [review]

**Review by DonovanDMC — APPROVED** [review]

**Pull request merged**